psi4/driver/procrouting/proc.py

```diff
@@ -1788,6 +1794,9 @@ def run_dfocc_gradient(name, **kwargs):
 
     proc_util.check_disk_df(name.upper(), optstash)
 
+    if core.get_global_option('SCF_TYPE') != 'DISK_DF':
```

Comment:
I always wondered why `MEM_DF` is not allowed here.

Reply:
`MEM_DF` doesn't save the DF integrals for later use.

Reply:
Small correction. MEM_DF can and does use disk for 3-electron integrals in its disk mode (again I really regret these names). The integrals are dumped in a format only usable by the MEM DF object. It is a pretty complex packing where the sparse tensors are organized into a dense format for GEMM.

Reply:
true, i know about the disk-mode in `mem_df`, though no details. I only meant to say that there is currently no logic to keep these integrals around after the SCF is done, not that it wouldn't be possible.